pkg/products/rhssouser/reconciler.go

```diff
@@ -226,33 +315,364 @@ func (r *Reconciler) reconcileComponents(ctx context.Context, installation *inte
 		}
 
 		kcr.Spec.InstanceSelector = &metav1.LabelSelector{
-			MatchLabels: getInstanceLabels(),
+			MatchLabels: getMasterLabels(),
 		}
-		kcr.Labels = getInstanceLabels()
+
+		kcr.Labels = getMasterLabels()
+
 		kcr.Spec.Realm = &keycloak.KeycloakAPIRealm{
-			ID:              keycloakRealmName,
-			Realm:           keycloakRealmName,
-			Enabled:         true,
-			DisplayName:     keycloakRealmName,
-			EventsListeners: []string{"metrics-listener"},
+			ID:          masterRealmName,
+			Realm:       masterRealmName,
+			Enabled:     true,
+			DisplayName: masterRealmName,
 		}
 
 		// The identity providers need to be set up before the realm CR gets
 		// created because the Keycloak operator does not allow updates to
 		// the realms
-		err = r.setupOpenshiftIDP(ctx, installation, kcr, serverClient)
+		err := r.setupOpenshiftIDP(ctx, installation, kcr, serverClient)
 		if err != nil {
 			return errors.Wrap(err, "failed to setup Openshift IDP for user-sso")
 		}
 
 		return nil
 	})
 	if err != nil {
-		return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("failed to create/update keycloak realm: %w", err)
+		return nil, fmt.Errorf("failed to create/update keycloak realm: %w", err)
 	}
 	r.logger.Infof("The operation result for keycloakrealm %s was %s", kcr.Name, or)
 
-	return integreatlyv1alpha1.PhaseCompleted, nil
+	return kcr, nil
+}
+
+func (r *Reconciler) createOrUpdateKeycloakAdmin(user keycloak.KeycloakAPIUser, inst *integreatlyv1alpha1.Installation, ctx context.Context, serverClient k8sclient.Client) (controllerutil.OperationResult, error) {
+	kcUser := &keycloak.KeycloakUser{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("generated-%v", user.UserName),
+			Namespace: r.Config.GetNamespace(),
+		},
+	}
+
+	return controllerutil.CreateOrUpdate(ctx, serverClient, kcUser, func() error {
+		ownerutil.EnsureOwner(kcUser, inst)
```

Comment:
@briangallagher This would result in a invalid OwnerReference, and we already fixed the rest of them on master, so can you please remove this line?